_drafts/gdb-python-api-pypi-packages.md

```diff
@@ -0,0 +1,450 @@
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload',
+ '/Library/Python/2.7/site-packages',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python',
+ '/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/']
+```
+
+It looks like GDB is using the macOS System Python version 2.7.10 and is pulling
+in packages from the system installation at
+`'/Library/Python/2.7/site-packages'`. This is _wrong_, as it should be looking
+in our virtual environment's `site-packages/` directory.
+
+So how do we get the `sys.path` value within GDB's Python updated so I can
+import PrettyTable?!
+
+### Setting `sys.path` within `~/.gdbinit`
+
+We need three things:
+
+1. Find a way to modify GDB on initialization
+2. Extract our virtual environment's `sys.path` values
+3. append these paths to GDB's Python `sys.path` value.
+
+#### 1. Modifying GDB on launch
+
+To modify every instance of GDB that is launched, we will edit our `~/.gdbinit` file. However, there are *many* other ways to modify how GDB is launched, and those are covered in detail [here](http://sourceware.org/gdb/onlinedocs/gdb/Startup.html) in GDB's documentation. In summary, here is the order of steps GDB takes during launch:
+
+- Read and execute the system `gdbinit` file located at `/usr/share/gdb/gdbinit` or similar
+- Read and execute `~/.gdbinit`
+- Execute commands provided by args `-ix` and `-iex`
+- Read and execute the local `.gdbinit` file in the current directory, provided `set auto-load local-gdbinit` is set to `on`
+- Execute commands provided by args `-ex` and `-x`
+
+A tip to readers would be to use a combination of these setup steps, such as a project `.gdbinit` file that is committed to the repo so that a team can share common scripts!
+
+#### 2. Extract `sys.path` values from `venv`
+
+To output the value of `sys.path` in a shell, we can actually launch Python [with a given command](https://docs.python.org/2/using/cmdline.html#cmdoption-c) and have it print a value to standard out. Below, we import `sys`
+and immediately print `sys.path` to standard out.
+
+```python
+(venv)
+$ python -c "import sys;print(sys.path)"
+['/Users/tyler/venv/lib/python2.7/site-packages', ...]
+```
+
+Perfect! We can see our virtual environment's `site-packages/` directory in that
+list.
+
+#### 3. Append `sys.path` to GDB's Python
+
+Now we need to take the values from `sys.path` in step #2 and append them to
+GDB's `sys.path`
+
+In my local `~/.gdbinit` script, I will place the following code snippet at the
+bottom.
+
+```python
+# Update GDB's Python paths with the `sys.path` values of the local Python installation,
+#  whether that is brew'ed Python, a virtualenv, or another system python.
+
+# Convert GDB to interpret in Python
+python
+import sys
+import os
+import subprocess
+# Execute a Python using the user's shell and pull out the sys.path from that version
```

Comment:
maybe: might be able to format the comments here so you don't have to horizontal scroll on the rendered doc to read them

Reply:
Great idea!